lib/vector/Vlib/cats.c

```diff
@@ -493,7 +493,7 @@ int Vect_cat_list_to_array(const struct cat_list *list, int **vals, int *nvals)
 
     G_debug(1, "Vect_cat_list_to_array()");
 
-    *nvals = n_cats = 0;
+    *nvals = n_cats = n_ucats = 0;
     cats = NULL;
     for (i = 0; i < list->n_ranges; i++) {
```

Comment:
It seems to me to be a better solution to make an early exit before this for statement, with something like:
```c
if (list->n_ranges <= 0)
    return -1;
```

If `list->n_ranges` is 0 or less, `cats` and `n_cats` are never set... and the rest doesn't make any sense.
 
@metzm Perhaps you may have some insight in this?

Reply:
@nilason : Thanks for the review.

But, I am worried that '-1' indicates that there is something has gone wrong while converting using `Vect_cat_list_to_array` function, but here there is nothing wrong and it's just that the argument has no elements in it. What do you think about it?